### ISSUE TITLE
Change Chado loading code to handle genotype-genotype interactions . (genetic interactions)

### ISSUE THREAD

**Issue by kimrutherford** [pombase/pombase-chado#739]
The needs doing before we go live with pombase/canto#1915

**Comment by kimrutherford** [comment]
The code for this is now done and the tests pass.  I'm doing some local test loads on my desktop.  Still need to do pombase/website#1348 before we can try this out on the server.

**Comment by kimrutherford** [comment]
>  I'm doing some local test loads on my desktop.

That didn't go well.  I hadn't thought through all the cases properly.  I've tried a few options but I can't see a sensible way to store genotype-genotype interactions that have phenotypes in Chado.  It just not flexible enough.  I'll go back to the drawing board.

**Comment by ValWood** [comment]
;(

**Comment by kimrutherford** [comment]
> I hadn't thought through all the cases properly. I've tried a few options but I can't see a sensible way to store
> genotype-genotype interactions that have phenotypes in Chado.

I'm still not sure how to do this.  To make some progress I think we should keep Chado the same for now.  My suggestion is that we change the Chado loader to understand genotype-genotype interactions but (for now) discard the allele and genotype information and store a gene-gene genetic interaction exactly as we currently do.

That's not too much work and it will allow us to update the PomBase Canto to the genotype-genotype interaction version (like the test Canto).  And everything downstream, like the website code and exporting to BioGRID will continue to work.

**Comment by kimrutherford** [comment]
I've had an initial attempt at this.  It all seems to work fine for the data from the test Canto server.  Tomorrow I'll:

  - make a local copy of the main PomBase Canto database
  - upgrade that database to the new genotype-genotype interaction database format
  - export the sessions to a JSON file
  - do a full Chado load on my desktop including that JSON file

At that point I'll be able to compare the result of usual nightly load with my test.  My hope is that the output will be the same but we'll see.

**Comment by kimrutherford** [comment]
> Tomorrow I'll:

It worked pretty well.  In particular the number of genetic interactions in Chado is the same before and after changing Canto to genotype-genotype interactions.  I'm doing some deeper checking now.

**Comment by kimrutherford** [comment]
Postponed until https://github.com/pombase/canto/issues/2301 is complete.

**Comment by kimrutherford** [comment]
Based on our current discussion, these changes will need to be undone.

**Comment by kimrutherford** [comment]
> I'm going to try a local load with the new code tomorrow to see how far I get.

After a few attempts and a few fixes, I think it's all OK now.

**Comment by kimrutherford** [comment]
I've merged this code into the main branch so that it can be tested in the overnight load.  There are no genotype interactions yet but I want to make sure nothing else has been broken by the change.
